sympy/matrices/matrices.py

```diff
@@ -4150,35 +4252,129 @@ def LUdecomposition_Simple(self, iszerofunc=_iszero):
-                if pivot == -1 and not iszerofunc(A[i, j]):
-                    pivot = i
-            if pivot < 0:
-                # this result is based on iszerofunc's analysis of the possible pivots, so even though
-                # the element may not be strictly zero, the supplied iszerofunc's evaluation gave True
-                raise ValueError("No nonzero pivot found; inversion failed.")
-            if pivot != j:  # row must be swapped
-                A.row_swap(pivot, j)
-                p.append([pivot, j])
-            scale = 1 / A[j, j]
-            for i in range(j + 1, n):
-                A[i, j] = A[i, j] * scale
-        return A, p
+
+        if rankcheck:
+            SymPyDeprecationWarning(
+                feature="Keyword argument rankcheck is deprecated. "
+                        "Future LUdecomposition_Simple() will complete "
+                        "regardless rank of input matrix.",
+                issue=9796,
+                deprecated_since_version="1.1").warn()
+
+        if self.rows == 0 or self.cols == 0:
+            # Define LU decomposition of a matrix with no entries as a matrix
+            # of the same dimensions with all zero entries.
+            return self.zeros(self.rows, self.cols), []
+
+        lu = self.as_mutable()
+        row_swaps = []
+
+        pivot_col = 0
+        for pivot_row in range(0, lu.rows-1):
+            # Search for pivot. Prefer entry that iszeropivot determines
+            # is nonzero, over entry that iszeropivot cannot guarantee
+            # is  zero.
+            # XXX `_find_reasonable_pivot` uses slow zero testing. Blocked by bug #10279
+            # Future versions of LUdecomposition_simple can pass iszerofunc and simpfunc
+            # to _find_reasonable_pivot().
+            # In pass 3 of _find_reasonable_pivot(), the predicate in `if x.equals(S.Zero):`
+            # calls sympy.simplify(), and not the simplification function passed in via
+            # the keyword argument simpfunc.
+
+            candidate_pivot_row = None
+            while pivot_col != self.cols:
+                sub_col = (lu[r, pivot_col] for r in range(pivot_row, self.rows))
+                pivot_row_offset, pivot_value, is_assumed_non_zero, ind_simplified_pairs =\
+                    _find_reasonable_pivot_naive(sub_col, iszerofunc, simpfunc)
+                iszeropivot = pivot_value is None
+                if iszeropivot:
+                    # All candidate pivots in this column are zero.
+                    # Proceed to next column.
+                    pivot_col += 1
+                    continue
+
+                # Update entries simplified during pivot search.
+                for offset, val in ind_simplified_pairs:
+                    lu[pivot_row + offset, pivot_col] = val
+
+                candidate_pivot_row = None if pivot_row_offset is None else pivot_row + pivot_row_offset
+                break
```

Comment:
This is somehow confusing to me.  It would be more straight forward to say

```
while piv_col < self.cols:
   ...
   if pivot_value is not None:
       ....
       break
   pivot_cols += 1
```
and void the use of both a `continue` and a `break` in the same loop; especially a break that occurs at the end of the loop body.